pkg/controllers/reconcilers.go

```diff
@@ -323,6 +326,16 @@ func (r *ChildReconciler) Reconcile(ctx context.Context, parent apis.Object) (ct
 	child, err := r.reconcile(ctx, parent)
 	if err != nil {
 		if apierrs.IsAlreadyExists(err) {
+			// check if the failed create is owned by the parent.
+			// the created child from a previous turn may be slow to appear in the informer cache, but shouldn't appear
+			// on the parent as being not ready.
+			apierr := err.(apierrs.APIStatus)
```

Comment:
What ensures that this type assertion will not panic?

Reply:
the if check just above it